From d73b64632d1cb237209e765b3e597aa578a76d63 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Mon, 19 Jun 2023 08:25:42 +0200
Subject: [PATCH 0400/1141] drm/stm: lvds: reset during probing

The lvds must be reset during probing.

Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Change-Id: I84b67506790156f17f47a3148650fd5c2dfd0684
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/313526
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/stm/lvds.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

--- a/drivers/gpu/drm/stm/lvds.c
+++ b/drivers/gpu/drm/stm/lvds.c
@@ -19,6 +19,7 @@
 #include <linux/media-bus-format.h>
 #include <linux/module.h>
 #include <linux/of_device.h>
+#include <linux/reset.h>
 
 /* LVDS Host registers */
 #define LVDS_CR		0x0000  /* configuration register */
@@ -1069,6 +1070,7 @@ static int lvds_probe(struct platform_de
 	struct device_node *port1, *port2, *remote;
 	struct stm_lvds *lvds;
 	int ret, dual_link;
+	struct reset_control *rstc;
 
 	DRM_DEBUG_DRIVER("Probing LVDS driver...\n");
 
@@ -1105,6 +1107,17 @@ static int lvds_probe(struct platform_de
 		goto err_lvds_probe;
 	}
 
+	rstc = devm_reset_control_get_exclusive(dev, NULL);
+
+	if (IS_ERR(rstc)) {
+		ret = PTR_ERR(rstc);
+		goto err_lvds_probe;
+	}
+
+	reset_control_assert(rstc);
+	usleep_range(10, 20);
+	reset_control_deassert(rstc);
+
 	port1 = of_graph_get_port_by_id(dev->of_node, 1);
 	port2 = of_graph_get_port_by_id(dev->of_node, 2);
 	dual_link = drm_of_lvds_get_dual_link_pixel_order(port1, port2);
